File: 6_ Valid Parentheses/Stack/stack.cpp
#include <vector>
#include <iostream>

class Stack {
private:   
    std::vector<int> stack;
public:
    void push(int data){
        stack.push_back(data);
    }

    int pop() {
        if(!is_empty()) {
            int top = stack.back();
            stack.pop_back();
            return top;
        } else {
            std::cerr << "Stack is empty!" << std::endl;
            return -1;
        }
    }
    int peek() {
        if(!is_empty()) {
            return stack.back();
        } else {
            std::cerr << "Stack is empty!" << std::endl;
            return -1;
        }
    }

    bool is_empty()const {
        return stack.empty();
    }

};



int main(){
    Stack stack;
    stack.push(1);
    stack.push(2);
    stack.push(3);

    std::cout << stack.peek()<< std::endl; 
    std::cout << stack.pop()<< std::endl; 
    std::cout << stack.peek()<< std::endl; 
    std::cout << stack.pop()<< std::endl; 
    std::cout << stack.peek()<< std::endl; 

    return 0;
}

File: 6_ Valid Parentheses/Stack/stack.py
class Stack:
    def __init__(self):
        self.stack = []

    def push(self, value):
        self.stack.append(value)
    
    def pop(self):
        if not self.is_empty():
            return self.stack.pop()
        else:
            return "Stack is empty"

    def peek(self):
        if not self.is_empty():
            return self.stack[-1]
        else:
            return "Stack is empty"
        
    def is_empty(self):
        return len(self.stack) == 0   
    def size(self):
        return len(self.stack) 

stack = Stack()
stack.push(5)
stack.push(54)

print(stack.peek())
print(stack.pop())
print(stack.peek())